### PR TITLE
HBASE-26688 Threads shared EMPTY_RESULT may lead to unexpected client job down

### PR DESCRIPTION
**Pull request by YutSean**
https://issues.apache.org/jira/browse/HBASE-26688

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  1s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 35s |  master passed  |
| +1 :green_heart: |  compile  |   4m 33s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 44s |  master passed  |
| +1 :green_heart: |  spotbugs  |   3m 32s |  master passed  |
| -0 :warning: |  patch  |   2m 40s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 32s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 32s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 29s |  hbase-client: The patch generated 0 new + 11 unchanged - 2 fixed = 11 total (was 13)  |
| -0 :warning: |  checkstyle  |   1m  9s |  hbase-server: The patch generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  22m 11s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotbugs  |   3m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 23s |  The patch does not generate ASF License warnings.  |
|  |   |  62m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4072 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4dae62b12b3a 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / f6348d4100 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestResult.java [hunk 1]
```diff
@@ -131,18 +131,6 @@ public void testCurrentOnEmptyCell() throws IOException {
     assertNull(r.current());
   }
 
-  @Test
-  public void testAdvanceTwiceOnEmptyCell() throws IOException {
```
Comment (Apache9) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestResult.java:135]: Better keep this test but use two assertFalse to confirm that the second advance call will return false?

Comment (YutSean) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestResult.java:135]: > Better keep this test but use two assertFalse to confirm that the second advance call will return false?

After the change, empty list will always return false. I think we could change the UT's name to testAdvanceMultipieOnEmptyCell and use an arbitrary value > 2 to assert false. What do you think?

Comment (Apache9) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestResult.java:135]: It is also fine. Just go ahead.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 18s |  master passed  |
| +1 :green_heart: |  compile  |   1m 52s |  master passed  |
| +1 :green_heart: |  shadedjars  |   9m  6s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 10s |  master passed  |
| -0 :warning: |  patch  |  10m 37s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m  9s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 45s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 208m  5s |  hbase-server in the patch failed.  |
|  |   | 248m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4072 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ca28ce3918f0 4.15.0-153-generic #160-Ubuntu SMP Thu Jul 29 06:54:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / f6348d4100 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/testReport/ |
| Max. process+thread count | 3729 (vs. ulimit of 30000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 22s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  5s |  master passed  |
| +1 :green_heart: |  compile  |   1m 51s |  master passed  |
| +1 :green_heart: |  shadedjars  |   9m 58s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 12s |  master passed  |
| -0 :warning: |  patch  |  11m 32s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 47s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 47s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m 59s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 48s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 254m  6s |  hbase-server in the patch failed.  |
|  |   | 297m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4072 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 873c4c018138 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / f6348d4100 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/testReport/ |
| Max. process+thread count | 2892 (vs. ulimit of 30000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4072/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**